### PR TITLE
cherry-pick of #17692

### PR DESCRIPTION
**Pull request by kyessenov**
Manual cherry-pick of #17692. Automatic does not work due to the change from gogo to golang protos.
Updates proxy and api SHAs.

### CODE REVIEW

**Comment by kyessenov** [comment]
@rlenglet @mandarjog

File: go.sum [hunk 1]
```diff
@@ -815,6 +815,8 @@ istio.io/api v0.0.0-20190405142752-6b8d1849e7f4/go.mod h1:hhLFQmpHia8zgaM37vb2ml
 istio.io/api v0.0.0-20190515205759-982e5c3888c6/go.mod h1:hhLFQmpHia8zgaM37vb2ml9iS5NfNfqZGRt1pS9aVEo=
 istio.io/api v0.0.0-20190820204432-483f2547d882 h1:L0WC/5HTk8T5eGTg/ka9jGZgw7GMuWj9rm6DFF4owL8=
 istio.io/api v0.0.0-20190820204432-483f2547d882/go.mod h1:42cBjnu/rTJcCaKi8nLdIvq0n71RcLrkgZ9IQSvDdSQ=
+istio.io/api v0.0.0-20200124023137-04d3a3d0471f h1:vITUEcljxMwz2Czn1SfeJLr/+RAcwOtzGtNjd7mzKTQ=
```
Comment (rlenglet) [inline go.sum]: nit: You should be able to remove the older versions of `istio.io/api`. Just delete the 4 lines above.

**Review by rlenglet — APPROVED** [review]

**Comment by rlenglet** [comment]
It seems like this needs a rebase.

**Comment by kyessenov** [comment]
Missed an updated test from #17925

**Pull request merged**